### PR TITLE
Upgrade to ES 5.2.2

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: app/src/main/java/io/crate/node/CrateNode.java [hunk 1]
```diff
@@ -25,7 +25,11 @@
 import com.google.common.collect.ImmutableList;
 import io.crate.Build;
 import io.crate.Version;
-import io.crate.plugin.*;
```
Comment (dobe) [inline app/src/main/java/io/crate/node/CrateNode.java]: do we want such formatting only changes?

File: blackbox/docs/sql/analyzer.txt [hunk 2]
```diff
@@ -533,6 +533,16 @@ Converts alphabetic, numeric, and symbolic Unicode characters which are not in
 the first 127 ASCII characters (the "Basic Latin" Unicode block) into their
 ASCII equivalents, if one exists.
 
+.. _flattengraph-tokenfilter:
+
+``flatten_graph``
+-----------------
+
+``type='flatten_graph'``
```
Comment (dobe) [inline blackbox/docs/sql/analyzer.txt]: pls document this a bit more and add some links, i think for an end user this is not enough

File: core/src/main/java/io/crate/Version.java [hunk 3]
```diff
@@ -41,7 +41,7 @@
 
 
     public static final boolean SNAPSHOT = true;
-    public static final Version CURRENT = new Version(2010099, SNAPSHOT, org.elasticsearch.Version.V_5_1_2);
+    public static final Version CURRENT = new Version(2010099, SNAPSHOT, org.elasticsearch.Version.V_5_2_2_UNRELEASED);
```
Comment (dobe) [inline core/src/main/java/io/crate/Version.java]: why unreleased?

File: enterprise/users/build.gradle [hunk 4]
```diff
@@ -17,3 +17,18 @@ dependencies {
         exclude group: 'org.slf4j', module: 'jcl-over-slf4j'
     }
 }
+
+task writePropertiesFile {
+    doLast {
+        copy {
+            from "src/main/resources/"
+            into "${buildDir}/tmp"
```
Comment (dobe) [inline enterprise/users/build.gradle]: hm, is this a common pattern that we call this directory tmp? something like resources, expanded or built_templates would make it more informative

File: http/src/main/java/io/crate/rest/CrateRestMainAction.java [hunk 5]
```diff
@@ -119,29 +117,17 @@ void registerHandler() {
...
-                }
-            }
-        });
+        controller.registerHandler(GET, "/index.html", (req, channel, client) -> serveSite(siteDirectory, req, channel));
+
+        // FIXME: RestWrapper is only used if a Handler is registered, and wildcards are bound by separators
+        // This adds support for /static/{7-levels-deep-paths} - We should probably add an assertion to the downloadAdminUI task
+        // to make sure the adminUI doesn't use deeper paths
+        String staticPath = "/static";
+        for (int i = 97; i < 97 + 8; i++) {
```
Comment (dobe) [inline http/src/main/java/io/crate/rest/CrateRestMainAction.java]: try with /*/*/ or what about not using the rest handler?

File: sql/src/main/java/io/crate/operation/collect/sources/ShardCollectSource.java [hunk 6]
```diff
@@ -243,23 +244,48 @@ public void execute(Runnable command) {
         }
     }
 
+    private static class MemoizeSupplier<T> implements Supplier<T> {
```
Comment (dobe) [inline sql/src/main/java/io/crate/operation/collect/sources/ShardCollectSource.java]: add a note here about guava 19 suppliers

File: sql/src/main/java/io/crate/operation/collect/sources/ShardCollectSource.java [hunk 7]
```diff
@@ -243,23 +244,48 @@ public void execute(Runnable command) {
...
-                provider = new BlobShardCollectorProvider(blobShard, clusterService, nodeJobsCounter, functions,
-                    threadPool, settings, transportActionProvider);
-            } else {
-                provider = new LuceneShardCollectorProvider(
-                    schemas, luceneQueryBuilder, clusterService, nodeJobsCounter, functions,
-                    threadPool, settings, transportActionProvider, indexShard);
-            }
-            shards.put(indexShard.shardId(), provider);
+
+            /* The creation of a ShardCollectorProvider accesses the clusterState, which leads to a n
```
Comment (dobe) [inline sql/src/main/java/io/crate/operation/collect/sources/ShardCollectSource.java]: typo

File: core/src/main/java/io/crate/Version.java (hunk 3 above)
Comment (mfussenegger) [inline core/src/main/java/io/crate/Version.java]: To be honest I've no clue, everything after 5.0 has a _UNRELEASED suffix

**Pull request merged**